validator/project_validator.go

```diff
@@ -1111,6 +1113,29 @@ func validateTaskDependencies(project *model.Project) ValidationErrors {
 						project.Identifier, dep.Variant, task.Name),
 				})
 			}
+
+			dependent, exists := allTasks[dep.Name]
+			if !exists {
+				continue
+			}
+			if utility.FromBoolPtr(dependent.PatchOnly) {
```

Comment:
Sorry I didn't catch this already; we should check if the task is _not_ patch-only; otherwise this couldn't be scheduled at all for non-patches so it's a correct configuration (same for the other cases)

Reply:
I don't understand, that sounds like the case that this is trying to catch. If you depend on a task that is only sometimes able to be run, I want to warn because there will be scenarios where the upstream task won't have its dependencies satisfied

Reply:
What I mean is, if task A is patchable and depends on task B, and task B isn't patchable, then we should return a warning, because this is going to cause weird issues. But if task A and task B _both_ aren't patchable, then their configuration is fine and we shouldn't bother them with warnings.

Reply:
makes sense, fixed